portfolio/src/main/java/com/google/sps/servlets/DeleteMessageServlet.java

```diff
@@ -31,7 +31,20 @@
 public class DeleteMessageServlet extends HttpServlet {
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    long id = Long.parseLong(request.getParameter("id"));
+    UserService userService = UserServiceFactory.getUserService();
+    if (!userService.isUserLoggedIn()) {
+      response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+      return;
+    }
+
+    long id;
+    try {
+      id = Long.parseLong(request.getParameter("id"));
+    } catch (NumberFormatException e) {
+      System.err.println("Message id for deleting is not long: " + e);
```

Comment:
What do you mean by "deleting" here? You can also write this error description to the response to let the client know what the error is.